### PR TITLE
AMD introduction within Pal

### PR DESCRIPTION
**Pull request by jaysongiroux**
- amd/register/control_register/cr0.yml
     - Manual: https://www.amd.com/system/files/TechDocs/24593.pdf PAGE: 43

- amd/register/cpuid/CPUID_Fn00000000_EAX.yml
     - Manual: https://developer.amd.com/wp-content/resources/56255_3_03.PDF PAGE: 49

- amd/register/msr/MSR0000_0010.yml
     - Manual: https://developer.amd.com/wp-content/resources/56255_3_03.PDF PAGE: 77

- amd/register/vmcb/000h.yml
     - Manual: https://www.amd.com/system/files/TechDocs/24593.pdf PAGE 617

*This PR is subject to change as new yml files are added*

### CODE REVIEW

File: data/amd/register/msr/MSR0000_0010.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  access_mechanisms:
+      - name: rdmsr
+        is_read: True
+        address: 0x4c1
+
+      - name: wrmsr
+        is_write: True
+        address: 0x4c1
+
```
Comment (JaredWright) [inline data/amd/register/msr/MSR0000_0010.yml:18]: There are no fieldsets or fields defined for this MSR. According to the manual, it should contain one fieldset + 1 field named TSC (time stamp counter), that is 64 bits wide.

File: data/amd/register/msr/MSR0000_0010.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+       "
+       The TSC uses a common reference for all sockets, cores and threads. 
+       "
+  size: 64
+  arch: x86
+
+  access_mechanisms:
+      - name: rdmsr
+        is_read: True
+        address: 0x4c1
```
Comment (JaredWright) [inline data/amd/register/msr/MSR0000_0010.yml:13]: Where did you find the address for this MSR? I can't figure out how to find this in the manual on page 77, is it listed somewhere else?

**Review by JaredWright — CHANGES_REQUESTED** [review]
In addition to the changes requested, lets also go ahead and drop the template files for the time being. I think I'd like to remove those from the Intel folders as well eventually (not in this PR)

File: data/amd/register/cpuid/CPUID_Fn00000000_EAX.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+- name: CPUID_Fn00000000_EAX
+  long_name: "Processor Vendor and Largest Standard Function Number"
+  purpose: |
+    "
+    Processor Vendor and Largest Standard Function Number
+    "
+  size: 32
+  arch: x86
```
[the remark this answers is not in the document]
Comment (jaysongiroux) [reply data/amd/register/cpuid/CPUID_Fn00000000_EAX.yml:8]: thank you! took a bit to understand what to do with reserved bits. glad i was able to use intel's cr0 as an example.

File: data/amd/register/vmcb/000h.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  purpose: |
+      "
+      Intercept reads of CR0–15, respectively
+      "
+  size: 32
+  arch: amd64
+
+  access_mechanisms:
+      - name: read
+        offset: 000h
```
Comment (JaredWright) [inline data/amd/register/vmcb/000h.yml:12]: This line needs to be `offset: 0x0` or `offset: 0x000`

File: data/amd/register/vmcb/000h.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        component: vmcb
+
+      - name: write
+        offset: 000h
```
Comment (JaredWright) [inline data/amd/register/vmcb/000h.yml:16]: Same here

**Review by JaredWright — CHANGES_REQUESTED** [review]
Only two small things, then this looks ready to merge

File: data/amd/register/vmcb/000h.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    fields:
+      - name: "Bits 15_0"
+        long_name: "Intercept reads of CR0–15, respectively" 
+        lsb: 0
+        msb: 15
+        readable: True
+        writable: True
+
+      - name: "Bits 31_16"
+        long_name: "Intercept reads of CR0–15, respectively"
```
Comment (JaredWright) [inline data/amd/register/vmcb/000h.yml:33]: Just noticed this too: the manual describes bits 31:16 as "Intercept writes of CR0–15, respectively" (i.e. s/reads/writes)

**Review by JaredWright — APPROVED** [review]

**Pull request merged**